File: lib/UTS/models/restoran.js
import mongoose from "mongoose";

const RestoranSchema = new mongoose.Schema({
  nama: { type: String, required: true },
  alamat: { type: String, required: true },
  jarak: { type: Number, required: true },
  gambar: { type: String, required: true },
  menu: [{ type: mongoose.Schema.Types.ObjectId, ref: "Menu" }],
});

const Restoran = mongoose.model("Restoran", RestoranSchema);
export default Restoran;

File: lib/UTS/datas/restoran.dart
class Restaurant {
  final String id;
  final String nama;
  final String alamat;
  final String imageUrl;
  final List<String> menuIds;

  Restaurant({
    required this.id,
    required this.nama,
    required this.alamat,
    required this.imageUrl,
    required this.menuIds,
  });

  factory Restaurant.fromJson(Map<String, dynamic> json) {
    return Restaurant(
      id: json['_id'],
      nama: json['nama'],
      alamat: json['alamat'],
      imageUrl: json['gambar'],
      menuIds: List<String>.from(json['menu']),
    );
  }
}
